include/common/match.h

```diff
@@ -0,0 +1,12 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+#ifndef __LABWC_MATCH_H
+#define __LABWC_MATCH_H
+
+/**
+ * match() - Pattern match case-insensitively using '*' wildcards and '?' jokers
+ * @pattern: Pattern to match against.
+ * @string: String to search.
+ */
+bool match(const gchar *pattern, const gchar *string);
+
+#endif /* __LABWC_MATCH_H */
```

Comment:
This seems to miss an glib include due to the use of `gchar`. Otherwise it would always have to be included after something else which includes glib.

Reply:
Added `#include <glib.h>`

Regarding naming... :smile: I did actually call it `match-glob.c` and `match_glob()` - but changed when I realized that we'll have a clash with the osd-PR and I was keen to avoid a merge conflict (sorry for being so lazy)